test/access/roles/WhitelistAdminRole.test.js

```diff
@@ -2,21 +2,82 @@ const { shouldBehaveLikePublicRole } = require('../../behaviors/access/roles/Pub
+  '0xa30a26c630293F9dE3f35cDA36B24b5daeB69d01',
+  '0x3eA44e564D176c46ae29c549055012159494Ee48',
+  '0x9f2eC7D17A344E8326bCfaeA62d8Ac7D500b1348',
+  '0x768522fDF347c4aC88D4d78663710e3B98d5973C',
+  '0xB93c495880457A70A523cb702e8Eded3d11089ea',
+];
+
 contract('WhitelistAdminRole', function ([_, whitelistAdmin, otherWhitelistAdmin, ...otherAccounts]) {
   beforeEach(async function () {
     this.contract = await WhitelistAdminRoleMock.new({ from: whitelistAdmin });
     await this.contract.addWhitelistAdmin(otherWhitelistAdmin, { from: whitelistAdmin });
   });
 
   shouldBehaveLikePublicRole(whitelistAdmin, otherWhitelistAdmin, otherAccounts, 'whitelistAdmin');
-  it('removes role from all other accounts but the whitelister', async function () {
-    await this.contract.resetWhitelist({ from: whitelistAdmin });
-    await shouldFail.reverting(this.contract.onlyWhitelistAdminMock({ from: otherWhitelistAdmin }));
-    await this.contract.onlyWhitelistAdminMock({ from: whitelistAdmin });
+});
+
+contract('WhitelistAdminRole', function ([whitelistAdmin, otherWhitelistAdmin, ...otherAccounts]) {
+  it('removes role from all other accounts but the whitelister', async () => {
+    const contract = await WhitelistAdminRoleMock.new({ from: whitelistAdmin });
+    await contract.addWhitelistAdmin(otherWhitelistAdmin, { from: whitelistAdmin });
+    await contract.addWhitelistAdmin(otherAccounts[0], { from: whitelistAdmin });
+    await contract.resetWhitelist({ from: whitelistAdmin });
+    await shouldFail.reverting(contract.onlyWhitelistAdminMock({ from: otherWhitelistAdmin }));
+    await shouldFail.reverting(contract.onlyWhitelistAdminMock({ from: otherAccounts[0] }));
+    await contract.onlyWhitelistAdminMock({ from: whitelistAdmin });
   });
-  it('Transferring ownership should also add to whitelist', async function () {
+
+  it('Transferring ownership should also add to whitelist', async () => {
     const contract = await WhitelistAdminRoleMock.new({ from: whitelistAdmin });
     await contract.transferOwnership(otherWhitelistAdmin, { from: whitelistAdmin });
     await contract.onlyWhitelistAdminMock({ from: otherWhitelistAdmin });
   });
 });
+
+const add18Admins = contract => ADDRESSES.slice(0, 18).map(address => contract.addWhitelistAdmin(address));
+
+contract.only('WhitlistAdminRole', function ([deployer, secondAccount]) {
+  it('should not be possible to add more than 20 admins', async () => {
+    const contract = await WhitelistAdminRoleMock.new({ from: deployer });
+
+    await contract.addWhitelistAdmin(secondAccount);
+
+    const addAdressesToWhitelistAdminsPromises = add18Admins(contract);
+    await Promise.all(addAdressesToWhitelistAdminsPromises);
+
+    await shouldFail.reverting(contract.addWhitelistAdmin(ADDRESSES[18], { from: deployer }));
+    await shouldFail.reverting(contract.addWhitelistAdmin(ADDRESSES[19], { from: deployer }));
+  });
+
+  it('should be possible to add more admins after resetting', async () => {
```

Comment:
This test still fails. Resetting only sets the `role.bearer` from `true` to `false`. It does not resize the array.